src/vehicle.cpp

```diff
@@ -6545,7 +6545,12 @@ void vehicle::use_washing_machine( int p ) {
             n.set_age( 0 );
         }
 
-        drain( "water", 24 );
+        if ( fuel_left( "water" ) >= 24 ) {
+            drain( "water", 24 );
+        }
+        else {
+            drain( "water_clean", 24 );
```

Comment:
What if there is less than 24 charges of  `water_clean` here?

Reply:
For now this means that if there are less than 24 units of ordinary `water`, it drains water clean, no matter if there is clean water at all. There should be check for clean water, not just simple `else`.
Also, styling: should be `} else {`.

Reply:
The check for water amounts is done prior, this code can't be executed unless there is either more than 24 charges of water or clean water (or both). If there is less the "you need 24 charges of water" message is displayed. Or am I missing something?